compiler/noirc_frontend/src/lexer/token.rs

```diff
@@ -126,7 +126,7 @@ pub enum BorrowedToken<'input> {
     Invalid(char),
 }
 
-#[derive(PartialEq, Eq, Hash, Debug, Clone, PartialOrd, Ord)]
+#[derive(PartialEq, Eq, Hash, Debug, Clone, PartialOrd, Ord, Default)]
```

Comment:
Ah, good point. I added this in the very beginning but ended up not needing it. I'll remove it 👍

Reply:
Oh, it's because I use `std::mem::take` in one place with a `SpannedToken`... but I'll just create the "default" by hand, it'll be a bit more explicit and it's just in a few places.

Reply:
Done! I also found a way to do things a bit better, so now `std::mem::take` is never used, and `std::mem::replace` is used instead (I think that should be slightly faster, though I didn't notice any improvement in the big file I'm trying it with).